### PR TITLE
Migrate String#parseInt->Int|throws to String#toInt->Option[Int] ...

### PR DESCRIPTION
**Pull request by soc**
... same with String#parseLong.

### CODE REVIEW

File: tests/parse1.dora [hunk 1]
```diff
@@ -1,6 +1,6 @@
 fun main() {
-  assert("1".parseInt() == 1);
-  assert("100".parseInt() == 100);
-  assert("789".parseInt() == 789);
-  assert("bla".parseInt() == 0);
+  assert("1".toInt().unwrap() == 1);
+  assert("100".toInt().unwrap() == 100);
+  assert("789".toInt().unwrap() == 789);
+  assert("bla".toInt().isNone());
```
Comment (soc) [inline tests/parse1.dora]: This doesn't work:

```
#> target/debug/dora tests/parse1.dora
uncaught exception
3: String.toIntOrThrow() -> Int: 27
2: String.toInt() -> Option[Int]: 30
1: main(): 5
```

Any idea why?

File: dora/src/stdlib.rs [hunk 2]
```diff
@@ -249,18 +249,28 @@ pub extern "C" fn argv(ind: i32) -> Ref<Str> {
...
     let slice = val.content();
     let val = str::from_utf8(slice).unwrap();
 
-    val.parse::<i32>().unwrap_or(0)
+    let result = val.parse::<i32>();
+    if result.is_ok() {
+        return result.unwrap()
+    }
+    throw_native(true);
+    0
```
Comment (soc) [inline dora/src/stdlib.rs]: Can't use `unreachable!()`, this is actually reached!?

File: tests/parse1.dora (hunk 1 above)
Comment (dinfuehr) [reply tests/parse1.dora]: I think `toIntOrThrow` and `toLongOrThrow` need to be marked as `throws`. I would assume that this is the problem here, could also be a bug though.

File: dora/src/stdlib.rs (hunk 2 above)
Comment (dinfuehr) [reply dora/src/stdlib.rs]: Yes, throwing an exception from native code just sets a flag. When returning back to Dora code, there is assembly code that checks this flag and starts the exception handling process if it is set.

File: dora/src/stdlib.rs [hunk 3]
```diff
@@ -249,18 +249,28 @@ pub extern "C" fn argv(ind: i32) -> Ref<Str> {
...
+pub extern "C" fn str_to_long_or_throw(val: Ref<Str>) -> i64 {
     let slice = val.content();
     let val = str::from_utf8(slice).unwrap();
 
-    val.parse::<i64>().unwrap_or(0)
+    let result = val.parse::<i64>();
+    if result.is_ok() {
+        return result.unwrap()
+    }
+    throw_native(true);
```
Comment (dinfuehr) [inline dora/src/stdlib.rs]: `throw_native` is just a test function and isn't intended to be used apart from that. The boolean argument is just there to simulate a function that sometimes throws but also succeeds. Could you please create another function that just creates the exception object and sets the flag. throw_native could also use this function when the argument is true.

**Comment by dinfuehr** [comment]
Very cool, thanks! I think adding `throws` to the function definition should fix the error, then we can merge this. (`self.toIntOrThrow()` needs to be invoked as `try self.toIntOrThrow()` then as well)

File: tests/parse1.dora (hunk 1 above)
Comment (soc) [reply tests/parse1.dora]: Done, thanks!

File: dora/src/stdlib.rs (hunk 3 above)
Comment (soc) [reply dora/src/stdlib.rs]: Done!

**Comment by soc** [comment]
@dinfuehr Updated!

**Pull request merged**